packages/browser-wallet/src/popup/store/account.ts

```diff
@@ -24,7 +24,13 @@ export const selectedAccountAtom = atom<string | undefined, string | undefined>(
     (get) => get(storedAccountAtom),
     (_, set, address) => {
         set(storedAccountAtom, address);
-        popupMessageHandler.broadcast(EventType.AccountChanged, address);
+        popupMessageHandler.broadcast(EventType.AccountChanged, address, {
+            nonWhitelistedTabCallback: (tab: chrome.tabs.Tab) => {
+                if (tab.url) {
+                    popupMessageHandler.broadcastToUrl(EventType.AccountDisconnected, tab.url);
```

Comment:
Yeah.. so on second thought, this is not a good solution. Let's discuss this when you have the time (and also @abizjak's comment above)

Reply:
Changed this to:

- Removed disconnect emit from account change if not whitelisted
- Added emission of account change/disconnect to network change handler (if an account on the new network is connected, emit account change, otherwise account disconnect for the corresponding tab)